src/algokit/cli/dispenser.py

```diff
@@ -0,0 +1,202 @@
+    default="ci_token.txt",
+)
+def login_command(*, ci: bool, output: str) -> None:
+    if not ci and is_authenticated():
+        logger.info("Already logged in")
+        return
+
+    try:
+        audience = DispenserApiAudiences.CI if ci else DispenserApiAudiences.USER
+        custom_scopes = None if ci else "offline_access"
+        token_data = get_oauth_tokens(api_audience=audience, custom_scopes=custom_scopes)
+
+        if not token_data:
+            raise click.ClickException("Error obtaining auth token")
+
+        if ci:
+            token_file_path = output
+            with Path.open(Path(token_file_path), "w") as token_file:
+                token_file.write(token_data["access_token"])
+            logger.warning(
+                f"Your CI access token has been saved to `{token_file_path}`.\n"
+                "Please ensure you keep this file safe or remove after copying the token!"
+            )
+        else:
+            set_dispenser_credentials(token_data)
+            logger.info("Logged in!")
+
+    except Exception as e:
+        logger.debug(str(e))
+        raise click.ClickException(str(e)) from e
+
+
+@dispenser_group.command("fund", help="Fund your wallet address with TestNet ALGOs.")
+@click.option("--wallet", "-w", required=True, help="Wallet address to fund with TestNet ALGOs.")
+@click.option("--amount", "-a", required=True, help="Amount to fund. Defaults to microAlgos.", default=1000000)
+@click.option(
+    "--whole-units",
+    "whole_units",
+    is_flag=True,
+    help="Use whole units instead of smallest divisible units (microAlgos). Disabled by default.",
+    default=False,
+)
+@click.option(
+    "--ci",
+    "ci",
+    is_flag=True,
+    default=False,
+    help="Enable/disable interactions with Dispenser API via CI access token.",
+)
+def fund_command(*, wallet: str, amount: int, whole_units: bool, ci: bool) -> None:
+    if not ci and not is_authenticated():
+        logger.error("Please login first")
+        return
+
+    default_asset = assets[DispenserAssetName.ALGO]
+    if whole_units:
+        amount = amount * (10**default_asset.decimals)
+
+    try:
+        response = process_dispenser_request(
+            url_suffix=f"fund/{assets[DispenserAssetName.ALGO].asset_id}",
+            data={"walletAddress": wallet, "amount": amount, "assetID": default_asset.asset_id},
+            ci=ci,
+            method="POST",
+        )
+    except Exception as e:
+        logger.debug("Error processing dispenser fund request: %s", e)
+        raise click.ClickException(str(e)) from e
+
+    if response:
+        logger.info(response.json()["message"])
+
+
+@dispenser_group.command("refund", help="Refund ALGOs back to the dispenser wallet address.")
+@click.option("--txID", "-t", "tx_id", required=True, help="Transaction ID of your refund operation.")
+@click.option(
+    "--ci", is_flag=True, default=False, help="Enable/disable interactions with Dispenser API via CI access token."
```

Comment:
@robdmoore @neilcampbell do we want completely automated handling of ci tokens ? I though a more explicit --ci flag is more flexible - there could be cases when you might be testing something locally while setting up CI where you might want to have an ability to switch between user vs env var mode. Of course the cost is slightly more verbose command but don't we want to be a bit more predictable/transparent since we are dealing with sensitive access tokens?

Reply:
Upon further review and discussion with Rob, we added a prefix to env var name and at this point I agree, might simplify the code and make it simpler to have this implicit logic. Latest commit removes all --ci flags except for --ci on login (which is used for different purposes)